web/packages/teleterm/src/ui/HeadlessAuthn/HeadlessPrompt/HeadlessPrompt.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2021 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import * as Alerts from 'design/Alert';
+import { ButtonIcon, Text, ButtonSecondary } from 'design';
+import DialogConfirmation, {
+  DialogContent,
+  DialogHeader,
+  DialogFooter,
+} from 'design/DialogConfirmation';
+import { Attempt } from 'shared/hooks/useAsync';
+import * as Icons from 'design/Icon';
+
+import { PromptWebauthn } from '../../ClusterConnect/ClusterLogin/FormLogin/PromptWebauthn';
+
+export type HeadlessPromptProps = {
+  clientIp: string;
+  onApprove(): Promise<void>;
+  onReject(): Promise<void>;
+  updateHeadlessStateAttempt: Attempt<void>;
+  onCancel(): void;
+};
+
+export function HeadlessPrompt({
+  clientIp,
+  onApprove,
+  onReject,
+  updateHeadlessStateAttempt,
+  onCancel,
+}: HeadlessPromptProps) {
+  const [waitForMfa, setWaitForMfa] = useState(false);
+
+  return (
+    <DialogConfirmation
+      dialogCss={() => ({
+        maxWidth: '480px',
+        width: '100%',
+      })}
+      disableEscapeKeyDown={false}
+      open={true}
+    >
+      <DialogHeader justifyContent="space-between" mb={0} alignItems="baseline">
+        <Text typography="h4" bold>
+          Headless login detected
+        </Text>
+        <ButtonIcon type="button" onClick={onCancel} color="text.slightlyMuted">
+          <Icons.Close fontSize={5} />
+        </ButtonIcon>
+      </DialogHeader>
+      {updateHeadlessStateAttempt.status === 'error' && (
+        <Alerts.Danger mb={0}>
+          {updateHeadlessStateAttempt.statusText}
+        </Alerts.Danger>
+      )}
+
+      {!waitForMfa && (
+        <>
+          <DialogContent>
+            <Text typography="body1" color="text.slightlyMuted">
+              Someone has initiated a command from {clientIp}. If it was not
+              you, click Reject and contact your administrator.
+            </Text>
+          </DialogContent>
+          <DialogFooter>
+            <ButtonSecondary
+              autoFocus
+              mr={3}
+              type="submit"
+              onClick={async e => {
+                e.preventDefault();
+                setWaitForMfa(true);
+                await onApprove();
+              }}
```

Comment:
It can be just (similarly for Reject):
```ts
onClick={e => {
  e.preventDefault();
  setWaitForMfa(true);
  onApprove();
}}
```

We don't do anything with the response from `onApprove`.